src/booking/containers/booking-wrapper/booking-wrapper.component.ts

```diff
@@ -0,0 +1,82 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { bookingBackButtonVisibilityStates, bookingCheckVisibilityStates, bookingDateVisibilityStates, bookingHomeVisibilityStates, bookingLocationVisibilityStates } from '../../constants/pageVisibilityStates';
+import { BookingNavigation, BookingNavigationEvent, BookingState, NewBooking } from '../../models';
+
+@Component({
+  selector: 'exui-booking-wrapper',
+  templateUrl: './booking-wrapper.component.html',
+  styleUrls: ['./booking-wrapper.component.scss']
+})
+export class BookingWrapperComponent implements OnInit {
+
+  public backVisibilityStates = bookingBackButtonVisibilityStates;
+  public bookingNavigationCurrentState: BookingState;
+  public bookingNavigationEvent = BookingNavigationEvent;
+  public bookingHomeVisibilityStates = bookingHomeVisibilityStates;
+  public bookingLocationVisibilityStates = bookingLocationVisibilityStates;
+  public bookingDateVisibilityStates = bookingDateVisibilityStates;
+  public bookingCheckVisibilityStates = bookingCheckVisibilityStates;
+  public navEvent: BookingNavigation;
+  public newBooking = {} as NewBooking;
+
+  constructor(private readonly router: Router) { }
+
+  public ngOnInit() {
+    this.bookingNavigationCurrentState = BookingState.HOME;
+  }
+
+  public onNavEvent(event: BookingNavigationEvent) {
+    this.navEvent = {
+      event,
+      timestamp: Date.now()
+    };
+    this.navigationHandler(event);
+  }
+
+  public navigationHandler(navEvent: BookingNavigationEvent) {
+    switch (navEvent) {
```

Comment:
for now it s ok but we break Open closed  and Seperation of Concerns , we'll finetune later by applying strategy pattern .